src/modules/endpoint/sagas/watchFetchEvent.js

```diff
@@ -1,42 +1,66 @@
 // @flow
 
 import type { Saga } from 'redux-saga'
-import { all, call, put, takeLatest } from 'redux-saga/effects'
-import type { FetchEventActionType, FetchEventFailedActionType, PushEventActionType } from '../../app/StoreActionType'
+import { all, call, put, select, takeLatest } from 'redux-saga/effects'
+import type {
+  FetchEventActionType,
+  FetchEventFailedActionType,
+  PushEventActionType,
+  PushEventLanguagesActionType
+} from '../../app/StoreActionType'
 import type { DataContainer } from '../DataContainer'
 import loadCityContent from './loadCityContent'
 import { ContentLoadCriterion } from '../ContentLoadCriterion'
+import isPeekingRoute from '../selectors/isPeekingRoute'
 
 function * fetchEvent (dataContainer: DataContainer, action: FetchEventActionType): Saga<void> {
   const { city, language, path, key, criterion } = action.params
   try {
-    const loadCriterion = new ContentLoadCriterion(criterion, false)
-    const cityContentLoaded = yield call(loadCityContent,
-      dataContainer, city, language,
-      loadCriterion
-    )
-    if (cityContentLoaded) {
-      // Only proceed if the content is ready to be pushed to the state. If not then the UI automatically displays an
-      // appropriate error
-      const [events, resourceCache, cityLanguages] = yield all([
+    const peeking = yield select(state => isPeekingRoute(state, { routeCity: city }))
+
+    const loadCriterion = new ContentLoadCriterion(criterion, peeking)
+    const languageValid = yield call(loadCityContent, dataContainer, city, language, loadCriterion)
+
+    // Only get languages if we've loaded them, otherwise we're peeking
+    const cityLanguages = loadCriterion.shouldLoadLanguages() ? yield call(dataContainer.getLanguages, city) : []
```

Comment:
This means we can close this issue, right? https://issues.integreat-app.de/browse/NATIVE-335